src/controllers/admin/api-key.ts

```diff
@@ -0,0 +1,420 @@
+import type { Request, Response } from 'express';
+import * as dotenv from 'dotenv';
+import { StatusCodes } from 'http-status-codes';
+import { check } from '../validator/index.js';
+import { validate } from '../validator/decorator.js';
+import { APIKeyService } from '../../services/admin/api-key.js';
+import type {
+	APIKeyCreateRequestBody,
+	APIKeyCreateResponseBody,
+	APIKeyCreateUnsuccessfulResponseBody,
+	APIKeyGetResponseBody,
+	APIKeyGetUnsuccessfulResponseBody,
+	APIKeyListResponseBody,
+	APIKeyListUnsuccessfulResponseBody,
+	APIKeyResponseBody,
+	APIKeyRevokeRequestBody,
+	APIKeyRevokeResponseBody,
+	APIKeyRevokeUnsuccessfulResponseBody,
+	APIKeyUpdateRequestBody,
+	APIKeyUpdateResponseBody,
+	APIKeyUpdateUnsuccessfulResponseBody,
+	APIServiceOptions,
+} from '../../types/portal.js';
+import { EventTracker, eventTracker } from '../../services/track/tracker.js';
+import { OperationNameEnum } from '../../types/constants.js';
+
+dotenv.config();
+
+export class APIKeyController {
+	static apiKeyCreateValidator = [
+		check('expiresAt')
+			.optional()
+			.isISO8601()
+			.toDate()
+			.withMessage('Invalid date format')
+			.custom((value) => {
+				if (value < new Date()) {
+					throw new Error('expiresAt must be in the future');
+				}
+				return true;
```

Comment:
AFAIK, `new Date()` will create a Date object based on the local timezone of the machine. Our server is in London, so it's actually just UTC for most of the year, except when it's summer time when it will be 1 hour out. The problem is more obvious if the server was in a completely different region/timezone, where this comparison could result in a failure.

You might need to something like the following (if you agree this could be a bug - maybe I'm misunderstanding this as something that could be a bug):

```typescript
.custom((value) => {
            const currentUtcDate = new Date(new Date().toISOString());
            if (value < currentUtcDate) {
                throw new Error('expiresAt must be in the future');
            }
            return true;
        })
```

Here, `toISOString` will create a normalised UTC timezone Date object

Reply:
Agreed!